### ISSUE TITLE
Unexpected output when combined with `child_process.fork()`

### ISSUE THREAD

**Issue by peterroe** [privatenumber/tsx#276]
### Bug description

Unexpected output when combined with `child_process.fork()`

### Reproduction

Situation :

```ts
// index.ts
import { fork } from 'child_process'
const child = fork('child.ts')
child.on('message', (data) => {
    console.log(`output: ${data}`);
});
```
```ts
// child.ts
process.send("hello")
```

run: 
```bash
$ tsx index.ts
output: [object Object]
output: hello
```
---

expected output
```bash
output: hello
```
actual output：

```bash
output: [object Object]. # redundant useless output
output: hello
```
---

env：
```
tsx v3.12.7
node v16.20.1
```

### Environment

```shell
System:
    OS: macOS 13.3
    CPU: (12) arm64 Apple M2 Pro
    Memory: 103.44 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.1 - ~/.nvm/versions/node/v16.20.1/bin/node
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.1/bin/npm
    pnpm: 6.35.1 - ~/.nvm/versions/node/v16.20.1/bin/pnpm
```

### Can you work on a fix?

- [X] I’m interested in opening a pull request to address this issue.

**Comment by peterroe** [comment]
Once I use `node` to execute `index.js` instead of `tsx`, there is no problem, so I suspect `tsx` problem

**Comment by privatenumber** [comment]
Does this still happen for you with the latest tsx? We made some fixes for Node ipc.

**Comment by peterroe** [comment]
Yes. it was fixed after I upgrade `tsx` to `4.6.2`
